publishing broke since a while

tests have been failing with e.g.

```
2023-08-06T06:12:47.5375621Z FAILED test_backups2datalad/test_core.py::test_1 - httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://api-staging.dandiarchive.org/api/dandisets/209734/versions/draft/publish/'
2023-08-06T06:12:47.5375795Z For more information check: https://httpstatuses.com/400
2023-08-06T06:12:47.5376370Z FAILED test_backups2datalad/test_core.py::test_2 - httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://api-staging.dandiarchive.org/api/dandisets/209738/versions/draft/publish/'
2023-08-06T06:12:47.5376543Z For more information check: https://httpstatuses.com/400
2023-08-06T06:12:47.5377227Z FAILED test_backups2datalad/test_core.py::test_3 - httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://api-staging.dandiarchive.org/api/dandisets/209744/versions/draft/publish/'
2023-08-06T06:12:47.5377491Z For more information check: https://httpstatuses.com/400
2023-08-06T06:12:47.5378062Z FAILED test_backups2datalad/test_core.py::test_4 - httpx.HTTPStatusError: Client error '400 Bad Request' for url 'https://api-staging.dandiarchive.org/api/dandisets/209747/versions/draft/publish/'
2023-08-06T06:12:47.5378221Z For more information check: https://httpstatuses.com/400

```

```shell
(venv) (dandisets) dandi@drogon:/mnt/backup/dandi/tinuous-logs/dandisets/2023$ grep -l '400 Bad Request' */*/github/cron/*/*/test.yml/*-*/3_test\ \(3.10\,\ test\).txt
07/23/github/cron/20230723T060343/1048da4/test.yml/438-failed/3_test (3.10, test).txt
07/30/github/cron/20230730T060344/3d2b364/test.yml/439-failed/3_test (3.10, test).txt
08/06/github/cron/20230806T060341/47a43a3/test.yml/440-failed/3_test (3.10, test).txt
08/13/github/cron/20230813T060331/7762b46/test.yml/441-failed/3_test (3.10, test).txt
08/20/github/cron/20230820T060319/46f9772/test.yml/442-failed/3_test (3.10, test).txt
```

looking at that heroku staging logs

```
023-08-06T06:07:53.654588+00:00 app[web.1]: 10.1.6.164 - - [06/Aug/2023:06:07:53 +0000] "GET /api/dandisets/209734/versions/draft/info/ HTTP/1.1" 200 2537 "-" "backups2datalad (https://github.com/dandi/dandisets) httpx/0.23.3 CPython/3.10.12"
2023-08-06T06:07:53.680418+00:00 app[web.1]: [06:07:53] WARNING  Bad Request:                                      log.py:241
2023-08-06T06:07:53.680420+00:00 app[web.1]: /api/dandisets/209734/versions/draft/publish/
2023-08-06T06:07:53.680830+00:00 app[web.1]: 10.1.6.164 - - [06/Aug/2023:06:07:53 +0000] "POST /api/dandisets/209734/versions/draft/publish/ HTTP/1.1" 400 50 "-" "backups2datalad (https://github.com/dandi/dandisets) httpx/0.23.3 CPython/3.10.12"
2023-08-06T06:07:53.747678+00:00 app[web.1]: 10.1.7.226 - - [06/Aug/2023:06:07:53 +0000] "GET /api/dandisets/209735/versions/draft/info/ HTTP/1.1" 200 2537 "-" "backups2datalad (https://github.com/dandi/dandisets) httpx/0.23.3 CPython/3.8.17"
2023-08-06T06:07:53.776063+00:00 app[web.1]: [06:07:53] WARNING  Bad Request:                                      log.py:241
2023-08-06T06:07:53.776064+00:00 app[web.1]: /api/dandisets/209735/versions/draft/publish/
2023-08-06T06:07:53.776609+00:00 app[web.1]: 10.1.7.226 - - [06/Aug/2023:06:07:53 +0000] "POST /api/dandisets/209735/versions/draft/publish/ HTTP/1.1" 400 50 "-" "backups2datalad (https://github.com/dandi/dandisets) httpx/0.23.3 CPython/3.8.17"
```

so not much of information.

--- a/tools/backups2datalad/adandi.py
+++ b/tools/backups2datalad/adandi.py
@@ -218,7 +218,7 @@ class RemoteDandiset(SyncRemoteDandiset):
         start = time()
         while time() - start < max_time:
             r = await self.aclient.get(f"{self.version_api_path}info/")
-            if r["status"] == "Valid":
+            if r["status"] == "Valid" and not r.get("asset_validation_errors"):
                 return
             await anyio.sleep(0.5)
         about = {

